modules/sdk/gradle-plugins-target-platform/src/main/java/com/liferay/gradle/plugins/target/platform/ResolveTask.java

```diff
@@ -0,0 +1,309 @@
+
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import org.gradle.StartParameter;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.invocation.Gradle;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.SkipWhenEmpty;
+import org.gradle.api.tasks.StopExecutionException;
+import org.gradle.api.tasks.TaskAction;
+
+import org.osgi.service.resolver.ResolutionException;
+
+/**
+ * @author Gregory Amerson
+ */
+public class ResolveTask extends DefaultTask {
+
+	public ResolveTask() {
+		final Project project = getProject();
+
+		File bndrunFile = project.file(
+			project.getBuildDir() + "/resolve.bndrun");
+
+		setBndrunFile(bndrunFile);
+	}
+
+	@OutputFile
+	public File getBndrunFile() {
+		return GradleUtil.toFile(getProject(), _bndrunFile);
+	}
+
+	@InputFiles
+	@SkipWhenEmpty
+	public FileCollection getBundles() {
+		return _bundles;
+	}
+
+	@InputFile
+	@SkipWhenEmpty
+	public File getDistro() {
+		return _distro.getSingleFile();
+	}
+
+	@InputFiles
+	@SkipWhenEmpty
+	public FileCollection getRequirements() {
+		return _requirements;
+	}
+
+	@Input
+	@Optional
+	public boolean isIgnoreFailures() {
+		return GradleUtil.toBoolean(_ignoreFailures);
+	}
+
+	@TaskAction
+	public void resolve() {
+		FileCollection requirements = getRequirements();
+
+		File distro = getDistro();
+
+		try {
+			_writeBndrunFile(requirements, distro, getBndrunFile());
+		}
+		catch (IOException ioe) {
+			StopExecutionException exception = new StopExecutionException();
+
+			exception.initCause(ioe);
+
+			throw exception;
+		}
+
+		File cnfDir = new File(getTemporaryDir(), Workspace.CNFDIR);
+
+		Project project = getProject();
+
+		project.mkdir(cnfDir);
+
+		Logger logger = getLogger();
+
+		try (Bndrun bndrun = Bndrun.createBndrun(null, getBndrunFile())) {
+			bndrun.setBase(getTemporaryDir());
+
+			Workspace workspace = bndrun.getWorkspace();
+
+			workspace.setBuildDir(cnfDir);
+
+			Gradle gradle = project.getGradle();
+
+			StartParameter startParameter = gradle.getStartParameter();
+
+			workspace.setOffline(startParameter.isOffline());
+
+			FileCollection bundles = getBundles();
+
+			FileSetRepository fileSetRepository = new FileSetRepository(
+				getName(), bundles.getFiles());
+
+			workspace.addBasicPlugin(fileSetRepository);
+
+			logger.info(
+				"Resolving runbundles required for {}",
+				bndrun.getPropertiesFile());
+
+			for (RepositoryPlugin repo : workspace.getRepositories()) {
+				repo.list(null);
```

Comment:
Sure you don't need the results of this method invocation?

Reply:
Well, the ResolveTask in bnd-gradle plugin doesn't use the results, so I assumed we wouldn't need it.